src/solvers/imex_ark2.jl

```diff
@@ -0,0 +1,233 @@
+export ClimaODEFunction, NewIMEXARKAlgorithm, NewARS343
+
+Base.@kwdef struct ClimaODEFunction{TL, TE, TI, L, D, S} <: DiffEqBase.AbstractODEFunction{true}
+    T_lim!::TL = nothing # nothing or (uₜ, u, p, t) -> ...
+    T_exp!::TE = nothing # nothing or (uₜ, u, p, t) -> ...
+    T_imp!::TI = nothing # nothing or (uₜ, u, p, t) -> ...
+    lim!::L = (u, p, t, u_ref) -> nothing
+    dss!::D = (u, p, t) -> nothing
+    stage_callback!::S = (u, p, t) -> nothing
+end
+
+# Don't wrap a ClimaODEFunction in an ODEFunction (makes ODEProblem work).
+DiffEqBase.ODEFunction{iip}(f::ClimaODEFunction) where {iip} = f
+DiffEqBase.ODEFunction(f::ClimaODEFunction) = f
+
+struct NewIMEXARKAlgorithm{VS <: StaticArrays.StaticArray, MS <: StaticArrays.StaticArray, NM} <: DistributedODEAlgorithm
```

Comment:
Once this passes tests, can we please remove all `New*` (and if we want to keep the old algorithms around for a bit, add `Old*` there--or better yet just archive those files).

Reply:
Ok, sounds good, maybe switching the old stuff to `Old` would be useful. This way we can merge and update while testing